controller/linodemachine_controller_helpers.go

```diff
@@ -107,20 +115,59 @@ func (r *LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineSc
 	return createConfig, nil
 }
 
-func buildInstanceAddrs(linodeInstance *linodego.Instance) []clusterv1.MachineAddress {
+func (r *LinodeMachineReconciler) buildInstanceAddrs(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope, instanceID int) ([]clusterv1.MachineAddress, error) {
 	addrs := []clusterv1.MachineAddress{}
-	for _, addr := range linodeInstance.IPv4 {
-		addrType := clusterv1.MachineExternalIP
-		if addr.IsPrivate() {
-			addrType = clusterv1.MachineInternalIP
-		}
+	ips, err := r.getInstanceIPv4Addresses(ctx, logger, machineScope, instanceID)
+	if err != nil {
+		logger.Error(err, "Failed to get instance ip addresses")
+		return nil, err
```

Comment:
nit:
```suggestion
		return nil, fmt.Errorf("get instance ips: %w", err)
```
For helper functions like this and in the case where you're both logging and returning an error, you can surface the context of an error with `fmt.Errorf`